compiler/semexprs.nim

```diff
@@ -2397,6 +2395,12 @@ proc semMagic(c: PContext, n: PNode, s: PSym, flags: TExprFlags): PNode =
   of mSizeOf:
     markUsed(c, n.info, s)
     result = semSizeof(c, setMs(n, s))
+  of mDefault:
```

Comment:
Why is this not in `magicsAfterOverloadResolution`?

Also why is this not handled in the `mDefault` implementations for the backends (`getNullValue` in vmgen, `createVar` in jsgen, I think `genObjectInit` in cgen)?